src/oic/oic/provider.py

```diff
@@ -568,7 +570,7 @@ def end_session_endpoint(self, request="", cookie=None, **kwargs):
             "endsession_endpoint"
         )().from_urlencoded(request)
 
-        logger.debug("End session request: {}", sanitize(esr.to_dict()))
+        logger.debug("End session request: {}".format(sanitize(esr.to_dict())))
```

Comment:
Logging should be lazy, no need to evaluate the expression if we are not going to log it.

Reply:
Changed it because one of the mypy or pylama, don't remember which one, complained about too many arguments.

Reply:
Just checked and it doesn't complain for me. Might be from an earlier iteration of the code?

Reply:
Might be. I'm not wed to one or the other version so I'll change back if that is what you want.

Reply:
Hm, looking at it, it was wrong previously as well. Probably a reason for the complaining.

It should be:
`logger.debug("End session request: %s", sanitize(esr.to_dict()))`

It is not possible to use `{}` in logging...

Reply:
Which is why I removed the `{}` in the latest commit.

Reply:
Yes, but you are missing the `%s` which will cause a `TypeError`. Also the `format()` is redundant.

Either do not change this at all and we will deal with it in separate issue, or add the `%s` to the message string.

Reply:
I'll add `%s` to the message string.